Fixing Multiple Issues in Select Element Rendering

blasferna opened the pull request:
This pull request addresses several issues related to the rendering of select element attributes. The identified problems are enumerated below and were fixed in this pull request:

1. **Missing `id` Attribute**
   The library fails to render the `id` attribute of the select element.   

   Fixed #125 


3. **Select option is inheriting attributes from the select**
   The options are rendering the attributes from the select when they shouldn't according to the widget definition.

   Fixed #132 


4. **Missing `multiple` and `selected` Attribute**
   The select element does not render the `multiple` and `selected` attribute.

   Fixed #119 


5. **Missing `required` Attribute**

   The `required` attribute of the widget is not being rendered.


6. **Custom Attributes for Options**

   Custom select widgets defining attributes in the `create_option` method or disabling some options do not render properly.

7. **Rendering of Grouped Select Options**

   Grouped options are currently rendered incorrectly.

   `Before`

   ![grouped_select](https://github.com/django-crispy-forms/crispy-tailwind/assets/8385910/e39f370f-007a-4b01-8952-3d8b31d990ac)

    `After`
 
    ![grouped_select_after](https://github.com/django-crispy-forms/crispy-tailwind/assets/8385910/5749a948-084a-4fc3-ad1b-47c4366843b5)


8. **Django Autocomplete Light Attributes**
   
    Custom selects that override the `build_attrs` method of the `Widget` object are not properly supported, leading to integration issues with other packages.

   `Before`
   
   The dal attributes are not rendered and because of that the select is not working properly

   ![dal_select_before](https://github.com/django-crispy-forms/crispy-tailwind/assets/8385910/9366cbc7-56f7-4469-ada7-f66f373abeba)


   `After`

   ![image](https://github.com/django-crispy-forms/crispy-tailwind/assets/8385910/d5fff715-25e2-430e-abd9-c94979619d3d)

    Fixed #136 
   
Also, this pull requests fixes #98 

There is a [repository](https://github.com/blasferna/crispy-tailwind-select-issue) where you can check the select with the fixes applied in this pull request.

blasferna commented:
@GitRon and @smithdc1, could you please review this pull request when you have a moment and let me know if there's anything missing to move forward with it? Thank you.

GitRon commented:
Hi @blasferna! Sorry, didnt get an email about it from Gitlab! I'll have a look as soon as I find the time! Thx for your effort!

GitRon commented:
Thx @blasferna, your changes look good!

@carltongibson @smithdc1 Whats the best way here for content review? How do you do this usually? 

Best regards  
Ronny

smithdc1 commented:
Hi @GitRon 

My currently preferred approach is to use the outputs from the results folder. With the [tailwind playground](https://play.tailwindcss.com/) it's fairly easy to paste the new output files there to see how it looks. 

It seems that there's not currently enough test coverage to see all the changes included in this PR. I can't see any changes in the tests for `multiple`, `selected` or `required`, for example? 🤔

GitRon commented:
Hi @smithdc1!

Where would I find this results folder?

And the issue about missing test cases were feedback for @blasferna, am I correct?

blasferna commented:
Hi @GitRon and @smithdc1,

Just wanted to let you know that I've addressed the feedback regarding test coverage for the select attributes in the PR. I've added tests for multiple, selected, and required attributes.

Let me know if there's anything else I can do to move this forward.

Best regards,
Blas

GitRon commented:
@blasferna Amazing! I just saw the "results" directory.

The output on the tailwind playground looks good:

![grafik](https://github.com/django-crispy-forms/crispy-tailwind/assets/3176075/7b38d392-9f06-4084-b6fe-57bd3e161d85)

@smithdc1 @carltongibson @justinmayer Do you folk want to have a look as well? Otherwise, I'd merge it in a couple of days.

And @blasferna, thx for all your effort! A very comprehensive description and clean code 👌

GitRon commented:
> @smithdc1 @carltongibson @justinmayer Do you folk want to have a look as well? Otherwise, I'd merge it in a couple of days.

No news is good news. Let's get it merged 💪

anorthall commented:
@blasferna This is incredible - thank you! <3
